src/Toml/Type/Printer.hs

```diff
@@ -171,6 +175,19 @@ prettyKeyValue options i = mapOrdered (\kv -> [kvText kv]) options . HashMap.toL
     showText :: Show a => a -> Text
     showText = Text.pack . show
 
+    showTextUnicode :: Text -> Text
+    showTextUnicode text = Text.pack quotedText
+      where
+        quotedText = show finalText
+        finalText = foldl (\acc (ch, asciiCh) -> acc ++ getCh ch asciiCh) "" asciiArr 
```

Comment:
`foldl` leaks memory and it's discouraged to use it. Instead, it's better to use `foldl'`